File: 2748/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(countBeautifulPairs([]int{2, 5, 1, 4}))
}

func countBeautifulPairs(nums []int) int {
	strs := []string{}
	ans := 0
	for _, num := range nums {
		strs = append(strs, strconv.Itoa(num))
	}
	for i := range len(strs) {
		for j := i+1; j < len(strs); j++ {
			first := strs[i][0] - '0'
			last := strs[j][len(strs[j])-1] - '0'
			if gcd(int(first), int(last) % 10) {
				ans++
			}
		}
	}

	return ans
}

func gcd(a, b int) bool {
	for b != 0 {
		a, b = b, a % b
	}
	return a == 1
}

File: 2748/main.rs
struct Solution{}


impl Solution {
    pub fn count_beautiful_pairs(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        let n = nums.len() as i32;
        for i in 0..n {
            let mut x = nums[i as usize];
            while x >= 10 {
                x /= 10;
            }
            for j in (i+1)..n {
                if Self::gcd(x, nums[j as usize] %10) {
                    res += 1;
                }
            }
        }
        res
    }

    fn gcd(mut a: i32, mut b: i32) -> bool {
        while b != 0 {
            let tmp = b;
            b = a % b ;
            a = tmp;
        }

        a == 1
    }
}

fn main() {
    println!("{}", Solution::count_beautiful_pairs(vec![2, 5, 1, 4]))
}
